src/runtime_src/tools/xclbin/XclBinClass.h

```diff
@@ -55,6 +55,10 @@ class XclBin {
   void setKeyValue(const std::string & _keyValue);
   void removeKey(const std::string & _keyValue);
 
+  public:
+    static void getKeyValueComponents(const std::string & _keyValue, std::string & _domain, std::string & _key,std::string & _value);
+    static std::string findKeyAndGetValue(const std::string & _searchDomain, const std::string & _searchKey, std::vector<std::string> _keyValues);
+
  public:
   Section *findSection(enum axlf_section_kind _eKind, const std::string _indexName = "");
```

Comment:
const std::string&

Reply:
In this case, it isn't possible for `findKeyAndGetValue() `to return a `const std::string &`.  This is due to the need to decode and encoded string into its individual values.

_Side Note_
Double checked the other `const std::string &` types and they are correct.  Note: `getKeyValueComponents()` last three arguments are returned by value.  I have added a comment to describe intended behavior.

Reply:
I am looking at the function signature of findSection 'const std::string foo = ""'  makes no sense to me.  Why cannot it not be &?

Reply:
My apology wasn't looking at the wrong line.
I updated the signature of the argument to be passed by reference and made the method const.

Reply:
Fixed